src/sentry/remote_config/storage.py

```diff
@@ -0,0 +1,115 @@
+from io import BytesIO
+from typing import TypedDict
+
+from sentry import options
+from sentry.models.files.utils import get_storage
+from sentry.models.projectkey import ProjectKey
+from sentry.utils import json
+
+JSONValue = str | int | float | bool | None | list["JSONValue"] | dict[str, "JSONValue"]
+
+
+class Options(TypedDict):
+    sample_rate: float
+    traces_sample_rate: float
+    user_config: JSONValue
+
+
+class StorageFormat(TypedDict):
+    options: Options
+    version: int
+
+
+class APIFormat(TypedDict):
+    id: str
+    sample_rate: float
+    traces_sample_rate: float
+    user_config: JSONValue
+
+
+class StorageBackend:
+    def __init__(self, key: ProjectKey) -> None:
+        self.driver = BlobDriver(key)
+        self.key = key
+
+    def get(self) -> APIFormat | None:
+        result = self.driver.get()
+        if result is None:
+            return None
+        return self._deserialize(result)
+
+    def set(self, value: APIFormat) -> None:
+        self.driver.set(self._serialize(value))
+
+    def pop(self) -> None:
+        self.driver.pop()
+
+    def _deserialize(self, result: StorageFormat) -> APIFormat:
+        assert self.key.public_key is not None
+        return {
+            "id": self.key.public_key,
+            "sample_rate": result["options"]["sample_rate"],
+            "traces_sample_rate": result["options"]["traces_sample_rate"],
+            "user_config": result["options"]["user_config"],
+        }
+
+    def _serialize(self, result: APIFormat) -> StorageFormat:
+        return {
+            "options": {
+                "sample_rate": result["sample_rate"],
+                "traces_sample_rate": result["traces_sample_rate"],
+                "user_config": result["user_config"],
+            },
+            "version": 1,
+        }
+
+
+class BlobDriver:
+    def __init__(self, project_key: ProjectKey) -> None:
+        self.project_key = project_key
+
+    @property
+    def key(self):
+        assert self.project_key.public_key is not None
+        return (
+            f"configurations/{self.project_key.project_id}/{self.project_key.public_key}/production"
+        )
+
+    @property
+    def storage(self):
+        return get_storage(self._make_storage_options())
+
+    def get(self) -> StorageFormat | None:
+        try:
+            blob = self.storage.open(self.key)
+            result = blob.read()
+            blob.close()
+        except Exception:
+            return None
+
+        if result is None:
+            return None
+        return json.loads(result)
+
+    def set(self, value: StorageFormat) -> None:
+        self.storage.save(self.key, BytesIO(json.dumps(value).encode()))
+
+    def pop(self) -> None:
+        try:
+            self.storage.delete(self.key)
+        except Exception:
+            return None
+
+    def _make_storage_options(self) -> dict | None:
```

Comment:
```suggestion
    def _make_storage_config(self) -> dict | None:
```
To match the name of the parameter for `get_storage`